src/kernel-install/kernel-install.c

```diff
@@ -143,9 +143,12 @@ static int context_copy(const Context *source, Context *ret) {
                 .entry_token_type = source->entry_token_type,
         };
 
-        copy.rfd = fd_reopen(source->rfd, O_CLOEXEC|O_DIRECTORY|O_PATH);
-        if (copy.rfd < 0)
-                return copy.rfd;
+        if (source->rfd != AT_FDCWD) {
+                copy.rfd = fd_reopen(source->rfd, O_CLOEXEC|O_DIRECTORY|O_PATH);
+                if (copy.rfd < 0)
+                        return copy.rfd;
+        } else
+                copy.rfd = AT_FDCWD;
```

Comment:
I prefer the following:

```c
if (source->rfd >= 0)
        fd_reopen(...);
else if (source->rfd == AT_FDCWD)
        copy.rfd = AT_FDCWD;
else
        assert_not_reached();

Reply:
still needs the error_checking for fd_reopen though.

Reply:
i agree with @YHNdnzj. I think we should not silently convert an invalid fd (i.e. one < 0 but != AT_FDCWD) into AT_FDCWD here, but rather hit an assert in that case. this is what @YHNdnzj is more or less proposing here

Reply:
tried another variant with initializing the copy with AT_FDCWD, just like the original. That makes the if condition prettier